sn_networking/src/record_store.rs

```diff
@@ -428,27 +428,41 @@ impl NodeRecordStore {
     /// Calculate the cost to store data for our current store state
     #[allow(clippy::mutable_key_type)]
     pub(crate) fn store_cost(&self, key: &Key) -> (NanoTokens, QuotingMetrics) {
-        let quoting_metrics = QuotingMetrics {
-            records_stored: self.records.len(),
-            max_records: self.config.max_records,
-            received_payment_count: self.received_payment_count,
-            live_time: self.timestamp.elapsed().as_secs(),
-        };
+        let live_time = self.timestamp.elapsed().as_secs();
+        let mut records_stored = self.config.max_records;
```

Comment:
prefer a default vaule of `0`, to avoid in case of zero cost and no distance set,
the metrics showing `stored_records` as the max_records.